qutip/tests/core/test_coefficient.py

```diff
@@ -116,10 +126,25 @@ def test_CoeffCallArgs(base, kwargs, tol):
     t = np.random.rand() * 0.9 + 0.05
     w = np.random.rand() + 0.5j
     val = np.exp(w * t * np.pi)
-    coeff = coefficient(f, **kwargs)
+    coeff = coefficient(base, **kwargs)
     assert np.allclose(coeff(t, {"w": w}), val, rtol=tol)
 
 
+@pytest.mark.parametrize(['base', 'tol'], [
+    pytest.param(h, 1e-10, id="func"),
+    pytest.param("a + b + t", 1e-10, id="string")
+])
+def test_CoeffCallArguments(base, tol):
+    # Partial args update
+    t = np.random.rand() * 0.9 + 0.05
```

Comment:
We should just us fixed test cases here instead of random numbers, I think. If we use random numbers the test might just pass or fail randomly which is bad, and it doesn't really buy us anything over picking some specific numbers I don't think. If we want to test a range of different numbers, we should explicitly test the numbers we want to.

As an aside, I'd like to add hypothesis to qutip tests at some point, especially stuff in the core, but that's a bit of a rabbit hole, so I'll leave it until after I've started and version 5 has settled down a bit more.

Reply:
Nathan asked me about this around the time I first introduced pytest, but I didn't know how to do it at the time.  I was basically learning how to do _any_ unit testing (no formal training strikes again!).  Seems like a good idea to me in general - random testing has bitten us a few times, although I did make some efforts to try and make sure that pytest reported _which_ random values were used so it could be reproduced.